fastly/block_fastly_service_v1_package.go

```diff
@@ -0,0 +1,107 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type PackageServiceAttributeHandler struct {
+	*DefaultServiceAttributeHandler
+}
+
+func NewServicePackage() ServiceAttributeDefinition {
+	return &PackageServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key: "package",
+		},
+	}
+}
+
+func (h *PackageServiceAttributeHandler) Register(s *schema.Resource) error {
+	s.Schema[h.GetKey()] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MaxItems: 1,
+		MinItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"filename": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				// sha512 hash of the file
+				"source_code_hash": {
+					Type:     schema.TypeString,
+					Optional: true,
```

Comment:
I'm personally curious why `source_code_hash` is `Optional` but `source_code_size` is not.

Reply:
source_code_size is now explicitly optional